### PR TITLE
Vulcan: Refactor anchor link on headings

### PR DESCRIPTION
**Pull request by ianrohde**
## Summary
I missed that we had similar functionality on the `Causes` section on `troubleshooting` Vulcan pages when working on https://github.com/iFixit/react-commerce/pull/1691

This circles back to clean things up by sharing the existing `id` targets, but carrying over the new naming that is more semantic.

<details>
  <summary>Video</summary>

https://github.com/iFixit/react-commerce/assets/1634505/581e8fc8-53a3-4a62-b27e-c5f53fcacb9d

</details>

## Modifications
1. [refactor HeadingSelfLink usage](https://github.com/iFixit/react-commerce/commit/6e140e3c7f21f6bcf4ba0ab33d9d631b49141926)
2. [add missing selfLinked props](https://github.com/iFixit/react-commerce/commit/730d693f6336047bcc7dd2ee5f300c2a58dc4917)

## CR/QA Notes
Tested on:
http://localhost:3000/Vulcan/Dryer_Not_Spinning
http://localhost:3000/Vulcan/Samsung_TV_Turns_On_By_Itself
qa_req 0

### CODE REVIEW

File: frontend/templates/troubleshooting/index.tsx [hunk 1]
```diff
@@ -128,7 +134,7 @@ function TableOfContents({ solutions }: { solutions: Section[] }) {
          {solutions.map((solution, index) => (
             <Link
                key={solution.heading}
-               href={`#solution-${index + 1}`}
+               href={`#${solution.heading.toLowerCase().replace(/\s/g, '-')}`}
```
Comment (danielbeardsley) [inline frontend/templates/troubleshooting/index.tsx:137]: I think we should extract this into a utils function: `solutionHeadingToId(solution.heading)` so we can be sure the same format is used in both places.

Comment (danielbeardsley) [reply frontend/templates/troubleshooting/index.tsx:137]: Looks like the value [ought to be url-encoded](https://stackoverflow.com/questions/20679500/how-to-encode-space-in-the-fragment-identifier-in-a-url) as well (to handle things besides spaces).

**Comment by danielbeardsley** [comment]
dev_block :100: 

Otherwise this looks good. CR :+1:

File: frontend/templates/troubleshooting/index.tsx (hunk 1 above)
Comment (ianrohde) [reply frontend/templates/troubleshooting/index.tsx]: https://github.com/iFixit/react-commerce/pull/1707/commits/b7107507ae64bd718acc49fe6085bfdb3d5eb4ee

File: frontend/templates/troubleshooting/solution.tsx [hunk 2]
```diff
@@ -206,9 +206,11 @@ export default function SolutionCard({
    index: number;
    solution: SolutionSection;
 }) {
+   const solutionHeadingId = solution.heading.toLowerCase().replace(/\s/g, '-');
```
Comment (danielbeardsley) [inline frontend/templates/troubleshooting/solution.tsx:209]: This also needs to use the new function I think.

Comment (ianrohde) [reply frontend/templates/troubleshooting/solution.tsx]: Good lookin' out 👍

**Review by danielbeardsley — APPROVED** [review]
CR :+1:

**Pull request merged**